### PR TITLE
fix #2565

### LINKED ISSUE #2565: Error appears after changing console username/password
**Describe the bug**
Error message appears after changing console username and password 
error message is `404 ProxyServlet: /admin/denali/xsrf` 

**To Reproduce**
1. Go to WebConsole
2. Change values for username and password
3. Click on Apply button
4. Click "Yes" to confirm changing

**Expected behavior**
Username and password should be changed without error appearing 

**Screenshots**
console_password

**Target Environment (please complete the following information):**
 - Board: Raspberry Pi 3 (also able to reproduce on RG10-20)
 - Kura version: 4.2-SNAPSHOT 

**Additional context**
- username and password are changed 
- there is no error in kura.log
- unable to reproduce on Kura 4.1

### PR DESCRIPTION
**Pull request by spnettec**
Signed-off-by: heyoulin <heyoulin@gmail.com>

Error appears after changing console username/password

**Related Issue:** #2565

### CODE REVIEW

**Comment by mstankovic** [comment]
@MMaiero @spnettec I checked this PR and i was unable to recreate #2565
i also checked few more scenarios with logging and changing username/password and everything works fine.

**Comment by spnettec** [comment]
@nicolatimeus  I think this bug has been fixed, please review it.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java [hunk 1]
```diff
@@ -372,80 +391,82 @@ private HttpContext initResourceContext(final HttpContext defaultContext) {
...
-        this.httpService.registerServlet(AUTH_RESOURCE_PATH, new SendStatusServlet(404), null, resourceContext);
-        this.httpService.registerServlet(CONSOLE_RESOURCE_PATH, new SendStatusServlet(404), null, resourceContext);
+        this.httpService.registerServlet("/", new RedirectServlet("/"::equals, () -> this.appRoot), null,
+                this.resourceContext);
+        this.httpService.registerServlet(AUTH_RESOURCE_PATH, new SendStatusServlet(404), null, this.resourceContext);
+        this.httpService.registerServlet(CONSOLE_RESOURCE_PATH, new SendStatusServlet(404), null, this.resourceContext);
 
         this.httpService.registerServlet(PASSWORD_AUTH_PATH,
-                new GwtPasswordAuthenticationServiceImpl(this.authMgr, CONSOLE_PATH), null, sessionContext);
+                new GwtPasswordAuthenticationServiceImpl(this.authMgr, () -> this.appRoot), null, this.sessionContext);
```
Comment (nicolatimeus) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:410]: The `appRoot` parameter was originally meant to allow having a 3rd party web ui available on the root. It is not related the console path. Maybe we can improve the parameter description to avoid misunderstanding. If you want to enable the customisation of the console path (which can be desirable) please introduce another configuration parameter.

**Review by nicolatimeus — CHANGES_REQUESTED** [review]
Hi @spnettec, the changes look good to me, just please address the comment below. Also please check why Eclipse CLA check is failing.

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java (hunk 1 above)
Comment (spnettec) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:410]: I don't understand what means this appRoot is. If change the appRoot this code
`
this.httpService.registerServlet("/", new RedirectServlet("/"::equals, this.appRoot), null, resourceContext);
`
will redirect console to the changed appRoot. That's why i thought appRoot is console path.

By the way i don't know why Eclipse CLA check is aways failing.

Comment (spnettec) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:410]: Should i change to this?
`this.httpService.registerServlet("/", new RedirectServlet("/"::equals, CONSOLE_PATH), null, resourceContext);`

If not next time the we can't enter console by root path '/'.

Comment (nicolatimeus) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:410]: I will try to explain better the use case for `appRoot`:

Suppose you want to add another web application to the gateway that is completely different from Kura web ui, for example a web application developed by a customer that manages only its application, and the customer wants it to be available on the root path `/`.

You can achieve this in the following way:

1. Create a custom bundle that serves the application on `/mycompany/myapp`.
2. Set the `appRoot` parameter to `/mycompany/myapp`.

In this way the custom application will be available on the root path, Kura console will be still available on `/admin/console` which is currently hardcoded.

In other words `appRoot` is simply the relative path to which the user will be redirected when accessing the root path, which is not necessarily the Kura console path.

Comment (spnettec) [reply kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/Console.java:410]: Thanks. I see.  You can review the change of #2568 about this and i'll close this PR and open other PR later because of Eclipse CLA failing

**Comment by spnettec** [comment]
Close this PR because of Eclipse CLA failing
